spec/dockly/docker_spec.rb

```diff
@@ -293,44 +302,51 @@
         end
       end
 
+      after do
+        image = ::Docker::Image.all.find do |image|
+          image.info['RepoTags'].include?('dockly_test:latest')
+        end
+        image.remove if image
+      end
+
       it 'builds a docker image' do
-        expect {
-          subject.generate!
-          File.exist?(docker_file).should be_true
-          Dockly::Util::Tar.is_gzip?(docker_file).should be_true
-          File.size(docker_file).should be > (1024 * 1024)
-          paths = []
-          Gem::Package::TarReader.new(gz = Zlib::GzipReader.new(File.new(docker_file))).each do |entry|
-            paths << entry.header.name
-          end
-          paths.size.should be > 1000
-          paths.should include('sbin/init')
-          paths.should include('lib/dockly.rb')
-          paths.should include('it_worked')
-        }.to change { ::Docker::Image.all(:all => true).length }.by(4)
+        expect { subject.generate! }.to change { ::Docker::Image.all(:all => true).length }.by(4)
+
+        expect(File.exist?(docker_file)).to be_true
+        expect(Dockly::Util::Tar.is_gzip?(docker_file)).to be_true
+        expect(File.size(docker_file)).to be > (1024 * 1024)
+        paths = []
+        Gem::Package::TarReader.new(Zlib::GzipReader.new(File.new(docker_file))).each do |entry|
+          paths << entry.header.name
+        end
+        expect(paths.size).to be > 1000
+        expect(paths).to include('sbin/init')
+        expect(paths).to include('lib/dockly.rb')
+        expect(paths).to include('it_worked')
       end
     end
 
     context 'when there is a registry' do
       subject {
         Dockly::Docker.new do
-          registry_import 'nahiluhmot/base', :tag => 'latest'
```

Comment:
Not necessary, but if you want to minimize space you can use [scratch](https://registry.hub.docker.com/_/scratch/).